### ISSUE TITLE
Final documentation improvements

### ISSUE THREAD

**Issue by choldgraf** [matplotlib/matplotlib#8555]
After many weeks of work I think that the documentation is in a much better state now. I think that we are 80% of the way towards considering this "good enough" for now. This issue is a place to keep some lingering issues / ideas that could be implemented before 2.1 is released:

# ToDo
- [ ] Cycler documentation
Use of the cycler should be demoed more extensively in the examples, and should also be introduced and explained in an introductory tutorial. See #8885 

- [ ] ~~The [developers guide](http://matplotlib.org/devdocs/devel/index.html) should be updated and made more clear.~~ outside the scope of this issue I think

- [x] [This blog post](http://pbpython.com/effective-matplotlib.html) or something like it should be converted into a "getting started with matplotlib" kind of thing. Maybe we should reach out to the author and ask them to submit it as a sphinx-gallery python file.

- [x] We need to go through all the open PRs that were working on sphinx gallery and make sure that they get either closed because the work has already been done, or merged soon otherwise they'll quickly become forgotten PRs.

# In progress

- [x] On that note the toolkits API pages should be generated with Autosummary so that we get auto-linking with the SG examples. (https://github.com/matplotlib/matplotlib/pull/8573)
- [x] [toolkits](http://matplotlib.org/devdocs/mpl_toolkits/index.html) should be sphinx-galleryified and put into their own folder within tutorials, or their own gallery. (https://github.com/matplotlib/matplotlib/pull/8573)
- [ ] We should organize the examples folder so that the "most important" folders come first. We'll either need to wait for SG to support this, or add numbering to some examples folders. (being addressed partially here: https://github.com/sphinx-gallery/sphinx-gallery/pull/233)
- [x] Document the `data` kwarg, show how it is used in a tutorial and show some examples that use it. Ideally using pandas dataframes (I know it adds a dependency but I guarantee that users want to know how to do this) (#8566)
- [x] Improve the screenshots page
From @story645: *I've been wanting a simple gallery of sorts that goes over the basic plot types - line, scatter, bar, pie, heatmap - in like 5 lines or less w/ nothing fancy and I think that'd be perfect for a tutorial type section.*
This is quite similar to the [screenshots page](http://matplotlib.org/devdocs/users/screenshots.html), which is currently buried and kinda hard to find. IMO we should turn this into a `plot types` folder in the tutorials section, and have one short example for each plot type. (#8581 for starters)
- [x] The PNGs generated for the front page should be instead pulled from the SG thumbnails, and clicking each one should take you to that example, not to the screenshots page. (#8581)
- [x] Make sure that plotting with categorical values is clearly documented as it's being released with 2.1 (also here: https://github.com/matplotlib/matplotlib/issues/7155 and taken care of in #8581 )

I think this will go a long way towards bringing MPL's documentation into the 21st century.

As a general thought, any time somebody writes a blog post that makes an incorrect assumption about matplotlib, e.g. [this one](https://dsaber.com/2016/10/02/a-dramatic-tour-through-pythons-data-visualization-landscape-including-ggplot-and-altair/#comment-32), we should take a hard look at the docs and figure out how to make that feature/ use-case easier to find. In addition, any time somebody writes a clear, well-written blog post that is better at explaining things than the MPL docs, we should improve the docs or ask that person to contribute their post as a tutorial.

Let me know if there are any other suggestions for this issue and I can append to the list. Is there any ETA on 2.1? I wanna make sure this is a realistic goal.

@story645 @NelleV @tacaswell @phobson @anntzer @dopplershift @QuLogic

**Comment by story645** [comment]
1. We can also ask the [blog post](http://pbpython.com/effective-matplotlib.html) author for permission and the make it an open new-contributor friendly issue. 
2. Oh categorical...I mostly dunno where to put them. (That's a secondary part of my motivation for the simple gallery)
3. Screenshots is a little more complicated than the simple gallery I want, but like you said it can be refined since so much of that information is replicated elsewhere.

**Comment by choldgraf** [comment]
1. If you want, I can try to reach out to them.
2. I agree, this is the big challenge of MPL since there is *so much* you can do with it. I think this is one benefit of the tutorials section. It should convey the best-practices for using 80% of the most common tools in the package. Maybe that's a place to put it. That plus a simple example called "Plotting with categorical data" or something. Just so long as it's easy to `ctrl+f` or google index.
3. I agree. My 2 cents is that we should split up screenshots into a gallery folder like the one you describe, and another folder for "highlights of MPL" that's in the examples gallery. Then we should get rid of `screenshots`

**Comment by story645** [comment]
1. Please do!
4. don't know if data kwarg needs full blown tutorial, wonder if a simple example will do the trick. - Yes, I keep harping on this 'cause I think a lot of the examples lean towards showing all the things and so it can get hard to figure out how to do the 1 thing.

**Comment by choldgraf** [comment]
Totally agree w/ it not needing its own tutorial, maybe just a section of a tutorial that's updated to show off this usecase plus the quick example. I think an example is actually a better place to convey this info, but the examples page is *so long* that it's really hard to parse :-/

**Comment by story645** [comment]
Yeah, I think an upcoming documentation issue should be "gutting examples" ... 
An option for this specific case is putting the simple gallery at the top of the examples page so these sorta things (the ones users may not realize they need to control-f) are really easy to find.

**Comment by choldgraf** [comment]
yeah - I think that prioritizing the most useful / helpful / well-constructed examples at the top is a good start!

**Comment by choldgraf** [comment]
btw - I reached out to the author of that post on reddit, and he said he's more than happy (I believe he said he'd be honored actually ;-) ) to have us cannibalize some of his content for a tutorial

**Comment by choldgraf** [comment]
I'm going to close this, as I think that the bulk of the work is done and there are PRs/Issues for the other stuff.
